stdlib/public/Concurrency/Actor.cpp

```diff
@@ -1498,56 +1381,89 @@ void DefaultActorImpl::enqueueStealer(Job *job, JobPriority priority) {
 
 }
 
-// Called with actor lock held on current thread
-Job * DefaultActorImpl::drainOne() {
-  SWIFT_TASK_DEBUG_LOG("Draining one job from default actor %p", this);
-
+DefaultActorImpl::SimpleQueue DefaultActorImpl::collectJobs() {
```

Comment:
I don't understand why we're collecting all of the jobs into a separate queue instead of just (1) processing the incoming queue by enqueuing into the main queue and then (2) dequeueing a job off the main queue.

Reply:
Collecting jobs has a side effect of reversing jobs order. Atomic enqueueing adds them in LIFO order, with the head pointing to the last job. But collected jobs are returned in the correct order, with the head pointing to the oldest job.

Reply:
Also, separation of collection and enqueueing helps to lower contention. Before both collection and enqueueing were done between `_status().load()` and `_status().compare_exchange_weak()`, now it is only collection.